db/db.py

```diff
@@ -0,0 +1,68 @@
+import sqlite3
+
+
+class DbGateway:
+    def __init__(self, db_path):
+        self.db_path = db_path
+        self.conn = None
+
+    def create_connection(self) -> bool:
+        """ create a db connection to SQLite database """
+        if self.is_connected():
+            return False
+        try:
+            self.conn = sqlite3.connect(self.db_path)
+            return True
+        except sqlite3.Error as e:
+            print(e)
+        return False
+
+    def close_connection(self) -> bool:
+        if self.is_connected():
+            self.conn.close()
+            self.conn = None
+            return True
+        return False
+
+    def is_connected(self) -> bool:
+        return self.conn is not None
+
+
+class ImgGateway(DbGateway):
+    def create_table(self) -> bool:
+        """ create a table (if not exists) for storing image hashes """
+        if not self.is_connected():
+            return False
+        try:
+            c = self.conn.cursor()
+            c.execute(
+                '''CREATE TABLE IF NOT EXISTS images (hash TEXT UNIQUE)''')
+            return True
+        except sqlite3.Error as e:
+            print(e)
+        return False
+
+    def insert_hash(self, image_hash) -> bool:
```

Comment:
we could use the image path as you mentioned instead, as long as we can guarantee that it would always be unique.

as for insertion order we could pass into a serialized datetime object from into the db.

Reply:
i think we can make a change to the image getting algorithm to keep the path to image from mega, so then we enforce this guarantee

yeah i think if we can do that easily then this is a good improvement to make so we never have to worry about overuse of space (regardless of how long it might take to get there)